autohooks/precommit/run.py

```diff
@@ -141,7 +144,7 @@ def init(self, total: int) -> None:
         """
         self._progress.update(self._task_id, total=total)
 
-    def update(self, advance: Optional[int] = 1) -> None:
+    def update(self, advance: int = 1) -> None:
```

Comment:
I'm a bit unsure about changing the type of an parameter.  But since _progress.advance() needs an fixed float and update() already sets a default value for advance, i believe this might be the cleanest solution to fix an occuring mypy error (and the Optional[int] might simply be not necessary)

Reply:
Furthermore mypy has some more problems with this file:
`autohooks/precommit/run.py:200: error: Argument 1 to "ReportProgress" has incompatible type "rich.progress.Progress"; expected "autohooks.terminal.Progress"  [arg-type]`
`autohooks/precommit/run.py:213: error: "Progress" has no attribute "finish_task"  [attr-defined]`
I still strugle to understand the context in which these errors occure (I understand that Progress indeed does not have an attribute finish_task, but can't figure out why it even should have it). Some further insights would be really appreciated :)

Reply:
For me both versions of the update method are correct (because the called method has a default argument of 1 too) therefore the change is fine to fix a mypy issue.